core/src/main/java/io/bisq/core/network/CoreSeedNodesRepository.java

```diff
@@ -1,58 +1,37 @@
 package io.bisq.core.network;
 
-import com.google.common.collect.Sets;
-import com.google.inject.name.Named;
-import io.bisq.core.app.BisqEnvironment;
-import io.bisq.network.NetworkOptionKeys;
 import io.bisq.network.p2p.NodeAddress;
 import io.bisq.network.p2p.seed.SeedNodesRepository;
-import lombok.Getter;
-import lombok.extern.slf4j.Slf4j;
-import org.apache.commons.lang3.StringUtils;
 
-import javax.annotation.Nullable;
+import javax.annotation.concurrent.NotThreadSafe;
 import javax.inject.Inject;
-import java.util.Arrays;
-import java.util.List;
 import java.util.Set;
-import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
-@Slf4j
+import static io.bisq.core.network.DefaultNodeAddresses.DEFAULT_LOCALHOST_SEED_NODE_ADDRESSES;
+import static io.bisq.core.network.DefaultNodeAddresses.DEFAULT_TOR_SEED_NODE_ADDRESSES;
+
+@NotThreadSafe
```

Comment:
@ManfredKarrer is it ok that i marked the class as not thread safe? Actually it had been safe to access `seedNodeAddresses` from multiple threads before; but it hadn't been so for any other field of the class

Reply:
Yes you can mark it. 

In general most classes are living only in the context of a single user thread. 
GUI app:
```UserThread.setExecutor(Platform::runLater);```
or if not UI app (seed nodes):
```UserThread.setExecutor(Executors.newSingleThreadExecutor());```

There are only a few areas where multi threading is used: 
- Connection (P2P network)
- Tor startup
- Wallet encryption
- Local persistence
- Price request
- Fee request
- DAO/BSQ

From BitcoinJ we map to user thread as well.

Usual strategy is to map the result from the threaded operation to the user thread (via ``` UserThread.execute(() -> ....);``` so the clients can be sure that the callbacks are living in the same thread context. That model removes a lot of the threading complexity and isolates threading only to those areas where it is required.